File: ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 2
end

class Raindrops
  def self.convert(num)
    result = ''
    result << 'Pling' if (num % 3).zero?
    result << 'Plang' if (num % 5).zero?
    result << 'Plong' if (num % 7).zero?
    result.empty? ? num.to_s : result
  end
end

File: javascript/raindrops/raindrops.js
var Raindrops = function() {};

Raindrops.prototype.convert = function(input) {
  var output = '';
  
  if (input % 3 === 0) output += 'Pling';
  if (input % 5 === 0) output += 'Plang';
  if (input % 7 === 0) output += 'Plong';
  if (output === '') output += input;

  return output;
};

module.exports = Raindrops;
